includes/Modules/Analytics_4.php

```diff
@@ -2140,4 +2165,139 @@ private function filter_analytics_allow_tracking_disabled( $allowed ) {
+			function( GoogleAnalyticsAdminV1alphaAudience $audience ) {
+				$audience_item = array(
+					'name'        => $audience->getName(),
+					'displayName' => $audience->getDisplayName(),
+					'description' => $audience->getDescription(),
+				);
+
+				$audience_slug = $this->get_audience_slug( $audience );
+				$audience_type = $this->get_audience_type( $audience_slug );
+
+				$audience_item['audienceType'] = $audience_type;
+				$audience_item['audienceSlug'] = $audience_slug;
+
+				return $audience_item;
+			},
+			$audiences
+		);
+
+		$this->get_settings()->merge(
+			array(
+				'availableAudiences'             => $available_audiences,
+				'availableAudiencesLastSyncedAt' => time(),
+			)
+		);
+
+		return $available_audiences;
+	}
+
+	/**
+	 * Gets the audience slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param GoogleAnalyticsAdminV1alphaAudience $audience The audience object.
+	 * @return string The audience slug.
+	 */
+	private function get_audience_slug( GoogleAnalyticsAdminV1alphaAudience $audience ) {
+		$display_name = $audience->getDisplayName();
+
+		if ( 'All Users' === $display_name ) {
+			return 'all-users';
+		}
+
+		if ( 'Purchasers' === $display_name ) {
+			return 'purchasers';
+		}
+
+		$filter_clauses = $audience->getFilterClauses();
+
+		if ( $filter_clauses ) {
+			if ( $this->has_audience_site_kit_identifier(
+				$filter_clauses,
+				'new_visitors'
+			) ) {
+				return 'new-visitors';
+			}
+
+			if ( $this->has_audience_site_kit_identifier(
+				$filter_clauses,
+				'returning_visitors'
+			) ) {
+				return 'returning-visitors';
+			}
+		}
+
+		// Return an empty string for user defined audiences.
+		return '';
+	}
+
+	/**
+	 * Gets the audience type based on the audience slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $audience_slug The audience slug.
+	 * @return string The audience type.
+	 */
+	private function get_audience_type( $audience_slug ) {
+		if ( ! $audience_slug ) {
+			return 'USER_AUDIENCE';
+		}
+
+		switch ( $audience_slug ) {
+			case 'all-users':
+			case 'purchasers':
+				return 'DEFAULT_AUDIENCE';
+			case 'new-visitors':
+			case 'returning-visitors':
+				return 'SITE_KIT_AUDIENCE';
+		}
+	}
+
+	/**
+	 * Checks if an audience Site Kit identifier
+	 * (e.g. `created_by_googlesitekit:new_visitors`) exists in a nested array or object.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array|object $data The array or object to search.
+	 * @param mixed        $identifier The identifier to search for.
+	 * @return bool True if the value exists, false otherwise.
+	 */
+	private function has_audience_site_kit_identifier( $data, $identifier ) {
+		if ( is_array( $data ) || is_object( $data ) ) {
+			foreach ( $data as $key => $value ) {
+				if ( is_array( $value ) || is_object( $value ) ) {
+					// Recursively search the nested structure.
+					$result = $this->has_audience_site_kit_identifier( $value, $identifier );
+
+					if ( false !== $result ) {
+						return $result;
+					}
```

Comment:
On a minor stylistic note I think this would be slightly improved if it was a little terser:

```suggestion
					if ( $this->has_audience_site_kit_identifier( $value, $identifier ) ) {
						return true;
					}
```

Reply:
I agree. Updated, thanks!